core/storage/wal.rs

```diff
@@ -87,7 +85,8 @@ pub enum CheckpointStatus {
 impl Wal for WalFile {
     /// Begin a read transaction.
     fn begin_read_tx(&self) -> Result<()> {
-        self.min_frame.replace(*self.nbackfills.borrow() + 1);
+        let mut shared = self.shared.write().unwrap();
+        shared.min_frame = shared.nbackfills + 1;
```

Comment:
this isnt really related to multithreading cuz im just trying to understand stuff better, but `min_frame` isn't in the wal file format right? https://www.sqlite.org/walformat.html

it's always a derivation of `nbackfills + 1` right -- signifying what should be read from WAL and what from the main database file? if so, why not make it a function so we dont have unnecessary state?

don't have to change it here since it's not relevant to the PR, but just tryna get to the bottom of things

Reply:
min frame is the minimum frame to start reading from the WAL. This is because checkpointing might not checkpoint the whole WAL so you just update this in memory `min_frame` up to the last frame you were able to checkpoint. This is currently wrong but will be fixed later.